openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1553,16 +1553,21 @@ TCN_IMPLEMENT_CALL(jlong, SSL, makeNetworkBIO0)(TCN_STDARGS,
         return 0;
     }
 
-    if (max_bio_size < 0) {
-        tcn_ThrowException(e, "max_bio_size < 0");
+    if (maxInternalBIOSize < 0) {
```

Comment:
nit: most of the time this will not be violated ... I wonder if we can just reduce the amount of conditionals and combine the precondition check into a single conditional:

```c
if (maxInternalBIOSize < 0 || maxNetworkBIOSize < 0) {
  tcn_ThrowException(e, "maxInternalBIOSize and maxNetworkBIOSize both must be >= 0");
  return 0;
}
```